pkg/operator/ceph/object/objectstore.go

```diff
@@ -954,40 +957,109 @@ func zoneUpdateWorkaround(objContext *Context, expectedZone, gotZone map[string]
 	if err != nil {
 		return err
 	}
+
+	// check expected number of placements.
```

Comment:
Update the unit tests to cover these new cases?

Reply:
done